btr-api/src/btr_api/services/registries_search.py

```diff
@@ -0,0 +1,119 @@
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Manages registries search api interactions.
+
+NOTE: this is temporary as the trickle feed listener will handle this in the future.
+"""
+from http import HTTPStatus
+
+import requests
+from flask import Flask
+
+from btr_api.exceptions import ExternalServiceException
+from btr_api.models import Submission
+
+
+class RegSearchService:
+    """
+    A class that provides utility functions for connecting with the BC Registries search-api.
+    """
+    app: Flask = None
+    svc_url: str = None
+    timeout: int = None
+
+    def __init__(self, app: Flask = None):
+        """Initialize the registries search service."""
+        if app:
+            self.init_app(app)
+
+    def init_app(self, app: Flask):
+        """Initialize app dependent variables."""
+        self.app = app
+        self.svc_url = app.config.get('SEARCH_SVC_URL')
+        self.timeout = app.config.get('SEARCH_API_TIMEOUT', 20)
+
+    def update_business(self, submission: Submission, business: dict, token: str) -> requests.Response:
+        """Update businesses via the search-api."""
+        try:
+            business_identifier = business['identifier']
+            # collect current parties
+            parties = []
+            for person in submission.payload.get('personStatements', []):
+                party_name = ''
+                for name in person.get('names'):
+                    if name.get('type') == 'individual':  # expecting this to be 'individual' or 'alternative'
+                        party_name = name.get('fullName')
+                        break
+                if not party_name:
+                    self.app.logger.debug('Person names: %s', person.get('names'))
+                    self.app.logger.error('Error parsing SI name for %s', business_identifier)
+
+                parties.append({
+                    'id': f"{business_identifier}_{person['uuid']}",
+                    'partyName': party_name,
+                    'partyRoles': ['significant individual'],
+                    'partyType': 'person'
+                })
+
+            # make update call to reg search with headers + payload
+            payload = {
+                'type': 'partial',
+                'businesses': [{
+                    'id': business['identifier'],
+                    'parties': {'add': parties}
+                }]
+            }
+
+            headers = {'Authorization': 'Bearer ' + token, 'Content-Type': 'application/json'}
+            resp = requests.put(url=self.svc_url + '/internal/solr/import',
+                                json=payload,
+                                headers=headers,
+                                timeout=self.timeout)
+
+            if resp.status_code not in [HTTPStatus.OK, HTTPStatus.CREATED, HTTPStatus.ACCEPTED]:
```

Comment:
We are specifically expecting one of these, or any 2xx would suffice ?

Reply:
We are expecting HTTPStatus.ACCEPTED, but really any 2XX should be fine (in case its changed later). I'll update this to be `if not resp.ok`